src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+    }
+  ]
+}
+```
+
+Assuming you wish to deploy a simple _Hello World_ application, the YAML for such application should look similar to this:
+
+```yaml|highlight=2,23,34|title=hello-world.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+  labels:
+    track: canary
+spec:
+  selector:
+    matchLabels:
+      name: app
+  template:
+    metadata:
+      labels:
+        name: app
+    spec:
+      containers:
+      - name: app
+        image: learnk8s/app:1.0.0
+        ports:
+        - containerPort: 8080
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: app
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+  selector:
+    name: app
+---
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: app
+spec:
+  rules:
+  - http:
+    paths:
+    - backend:
+        serviceName: app
+        servicePort: 80
+      path: /
+```
+
+The definition is quite long, and it's easy to overlook how the components relate to each other.
+
+For example:
+
+- When should you use port 80 and when port 8080?
+- Should you create a new port for every Service so that they don't clash?
+- Do label names matter? Should it be the same everywhere?
+
+Before focusing on the debugging, let's recap how the three components link to each other.
+
+Let's start with Deployment and Service.
+
+## Connecting Deployment and Service
+
+The surprising news is that Service and Deployment aren't connected at all.
+
+Instead, the Service points to the Pods directly and skips the Deployment altogether.
+
+So what you should pay attention to is how the Pods and the Service are related to each other.
+
+You should remember three things:
+
+1. The Service selector should match the label of the Pod
+1. The Service `targetPort` should match the `containerPort` of the container inside the Pod
+1. The Service `port` can be any number. Multiple Service can use the same port because they have different IP addresses assigned.
+
+The following diagram summarises the how to connect the ports:
+
+```slideshow
+{
+  "description": "Connecting a Service and a Pod",
+  "slides": [
+    {
+      "image": "ports-1.svg",
+      "description": "Consider the following Pod exposed by a Service."
+    },
+    {
+      "image": "ports-2.svg",
+      "description": "When you create a Pod, you should define the port `containerPort` for each container in your Pods."
+    },
+    {
+      "image": "ports-3.svg",
+      "description": "When you create a Service, you can define a `port` and a `targetPort`. _But which one should you connect to the container?_"
+    },
+    {
+      "image": "ports-4.svg",
+      "description": "`targetPort` and `containerPort` should always match."
+    },
+    {
+      "image": "ports-5.svg",
+      "description": "If your container exposes port 3000, then the `targetPort` should match that number."
+    }
+  ]
+}
+```
```

Comment:
delete the first two slides again, the third slide is nice and good enough

optional: draw an app into the red box where there is some visual feedback (a funny ear?) to the fact that we are listening on 8080